src/pages/filter/ui/panels/function-panel/components/compound-request/compound-request.store.ts

```diff
@@ -0,0 +1,216 @@
+    )
+  }
+
+  public get resetValue(): string {
+    return this.cachedValues?.reset || ''
+  }
+
+  // request creation step by step
+  public handleSetSingleRequest(
+    requestBlockIndex: number,
+    currentSelectIndex: number,
+    target: HTMLSelectElement,
+  ): void {
+    const requestData = getRequestData(
+      target,
+      currentSelectIndex,
+      functionPanelStore.problemGroups,
+    )
+
+    const newRequest = Object.fromEntries(getSortedArray(requestData))
+
+    const newRequestCondition = cloneDeep(this.requestCondition)
+
+    newRequestCondition.forEach((item, index) => {
+      if (index === requestBlockIndex) {
+        item[1] = newRequest
+      }
+    })
+
+    this.sendRequestAsync(newRequestCondition)
+  }
+
+  // prepared request creation
+  public handleSetComplexRequest(
+    resetName: string,
+    activeRequestIndex: number,
+  ): void {
+    const resetRequestData = getResetRequestData(
+      resetName,
+      functionPanelStore.problemGroups,
+    )
+
+    const newRequest = Object.fromEntries(getSortedArray(resetRequestData))
+
+    const newRequestCondition = cloneDeep(this.requestCondition)
+
+    newRequestCondition.forEach((requestCondition, index) => {
+      if (index === activeRequestIndex) {
+        requestCondition[1] = newRequest
+      }
+    })
+
+    this.sendRequestAsync(newRequestCondition, resetName)
+  }
+
+  // add new request block
+  public handleRequestBlocksAmount(
+    type: string,
+    setActiveRequestIndex: (length: number) => void,
+    activeRequestIndex: number,
+  ): void {
+    if (type === 'ADD') {
+      const emptyBlock: [number, any] = [1, {}]
+      const newRequestCondition = [
+        ...cloneDeep(this.requestCondition),
+        emptyBlock,
+      ]
+
+      setActiveRequestIndex(newRequestCondition.length - 1)
+      this.sendRequestAsync(newRequestCondition, 'empty')
+    } else {
+      const newRequestCondition = cloneDeep(this.requestCondition).filter(
+        (_item, index) => index !== activeRequestIndex,
+      )
+
+      setActiveRequestIndex(newRequestCondition.length - 1)
+
+      this.sendRequestAsync(
+        newRequestCondition,
+        getResetType(newRequestCondition[newRequestCondition.length - 1][1]),
+      )
+    }
+  }
+
+  // change reuqest condition number
+  public handleRequestConditionNumber(
+    requestBlockIndex: number,
+    value: string,
+  ): void {
+    if (+value < 0) return
+
+    const newRequestCondition = cloneDeep(this.requestCondition)
+
+    newRequestCondition.map((item, index) => {
+      if (index === requestBlockIndex) {
+        item[0] = +value
+      }
+    })
+
+    this.sendRequestAsync(newRequestCondition)
+  }
+
+  // choose an active request block
+  public handleActiveRequest(
+    requestBlockIndex: number,
+    event: any,
+    setActiveRequestIndex: (index: number) => void,
+  ): void {
+    const classList = Array.from(event.target.classList)
```

Comment:
I think it's not a good choice to use HTML element className as business logic value. May be we should pass `shouldMakeActive` argument?